src/v2/Components/FlashBanner/index.tsx

```diff
@@ -1,43 +1,104 @@
+import { SystemQueryRenderer } from "v2/Artsy/Relay/SystemQueryRenderer"
+interface FlashBannerProps {
+  contentCode?: string
+  me?: {
+    canRequestEmailConfirmation: boolean
+  }
 }
 
 /**
- * The component responsible for selecting a flash message key from the `flash_message` url query param
+ * The component responsible for selecting a determining and displaying a flash message
+ *
  */
-export const FlashBanner: React.FunctionComponent<{
-  messageCode?: string
-}> = props => {
-  const [messageCode, setMessageCode] = React.useState(props.messageCode)
-
-  React.useEffect(() => {
-    if (!messageCode) {
-      const query = qs.parse(window.location.search.slice(1))
-      if (query["flash_message"]) setMessageCode(query["flash_message"])
-    }
-  }, [messageCode])
-
-  if (!messageCode) return null
-  const Message = messageByQueryParam[messageCode]
-  if (!Message) return null
+export const FlashBanner: React.FC<FlashBannerProps> = props => {
+  /**
+   * A map indexing keys (which may come from a contentCode prop, query string or
+   * logic internal to the component + its props) to banner content.
+   */
+  const contentMap: Record<string, string> = {
+    confirmed: "Your email has been confirmed.",
+    already_confirmed: "You have already confirmed your email.",
+    invalid_token: "An error has occurred. Please contact support@artsy.net.",
+    blank_token: "An error has occurred. Please contact support@artsy.net.",
+    expired_token: "Link expired. Resend verification email.",
+  }
+
+  let contentCode = props.contentCode
+
+  if (!contentCode) {
+    contentCode = qs.parse(window.location.search.slice(1))["flash_message"]
+  }
+
+  if (!contentCode) {
+    contentCode = props.me?.canRequestEmailConfirmation
+      ? "email_confirmation_cta"
+      : null
+  }
+
+  if (!contentCode) {
+    return null
+  }
+
+  let content
+
+  if (contentCode === "email_confirmation_cta") {
+    content = <EmailConfirmationCTA />
+  } else {
+    content = contentMap[contentCode]
+  }
+
+  if (!content) {
+    return null
+  }
 
   return (
     <Banner dismissable backgroundColor="black100">
       <Sans color="white100" size="3" lineHeight={1} weight="medium">
-        {typeof Message === "string" ? Message : <Message />}
+        <Flex justifyContent="center" alignItems="center">
+          {content}
+        </Flex>
       </Sans>
     </Banner>
   )
 }
+
+const TrackedFlashBanner = track({
+  context_module: Schema.ContextModule.FlashBanner,
+})(FlashBanner)
+
+export const FlashBannerQueryRenderer: React.FC = () => {
+  const didMount = useDidMount()
+  const { relayEnvironment } = useSystemContext()
+
+  // This component is client-only, skip during SSR
+  if (!didMount) {
+    return null
+  }
+
+  return (
+    <SystemQueryRenderer
+      environment={relayEnvironment}
+      query={graphql`
+        query FlashBannerQuery {
+          me {
+            canRequestEmailConfirmation
+            email
```

Comment:
Do we actually need to query `email` (and `id`) here? Not seeing them being used.